libs/remix-tests/src/testRunner.ts

```diff
@@ -228,20 +229,26 @@ export function runTest (testName: string, testObject: any, contractDetails: Com
             method.send(sendParams).on('receipt', (receipt) => {
                 try {
                     const time: number = (Date.now() - startTime) / 1000.0
-                    const topic = Web3.utils.sha3('AssertionEvent(bool,string)')
+                    const assertionEventHashes = assertionEvents.map(e => Web3.utils.sha3(e.name + '(' + e.params.join() + ')') )
                     let testPassed = false
-
                     for (const i in receipt.events) {
                         const event = receipt.events[i]
-                        if (event.raw.topics.indexOf(topic) >= 0) {
-                            const testEvent = web3.eth.abi.decodeParameters(['bool', 'string'], event.raw.data)
+                        const eIndex = assertionEventHashes.indexOf(event.raw.topics[0]) // event name topic will always be at index 0
+                        if (eIndex >= 0) {
+                            const testEvent = web3.eth.abi.decodeParameters(assertionEvents[eIndex].params, event.raw.data)
                             if (!testEvent[0]) {
+                                if(eIndex === 0) { // for 'Assert.ok' method
+                                    testEvent[2] = 'false'
```

Comment:
how can you be sure at this point that the returned value is `false`

Reply:
Because for `Assert.ok` it has only two params. `if (!testEvent[0])` this condition will be satisfied when we will get `false` while expecting `true`. So I explicitly put those parameters in the object to access it similarly afterwards.